test/tests.js

```diff
@@ -180,10 +180,10 @@ QUnit.test('undefined', function (assert) {
 
 QUnit.test('expires option as days from now', function (assert) {
 	assert.expect(1);
-	var expires = new Date();
-	expires.setDate(expires.getDate() + 1);
+	var days = 200;
+	var expires = new Date(new Date() * 1 + days * 864e+5);
```

Comment:
I still find this test very hard to comprehend the intentions behind it. I wonder if it makes sense to stub the variants we have (such as JavaScript `Date`) and return a fixed result if it's called a certain way?

```javascript
test('...', () => {
  window.Date = mockedDate;
  // ...
  // restore stub
})
```
Another option is to have our own "Expires" class that allows the expires to be defined like:

```javascript
Cookies.set('foo', { expires: Cookies.Expires.minutes(7) });
Cookies.set('foo', { expires: Cookies.Expires.days(24) });
```

This way we can stub the expires by passing a fake that conforms to the expected interface:

```javascript
Cookies.set('test-cookie', { expires: FakeExpires() });
```

One more option is to allow inserting a custom Date mechanism so that the developer can decide how to handle it. Also, the test can just verify if the function is being called with the desired argument, not if the `Date` is on the right state:

```javascript
Cookies.setDateMechanism((expiresAttributeValue) => {
  return new Date(new Date() * 1 + expiresAttributeValue * 864e+5);
});

...

Cookies.setDateMechanism((expiresAttributeValue) => {
  assert.strictEqual(expiresAttributeValue, 5);
  return defaultDateMechanism;
});
```

Reply:
I'm starting to wonder if maintaining this `expires: Number` thing is worth it and we should instead deprecate it to allow each developer to define their own way to handle dates, without being coupled to JavaScript `Date` class.

Reply:
It is worth rethinking the approach, no question, though of all the options I still find the native `Date` the most compelling one, due to its simplicity (no need for extra implementation, no need for a user to learn a new concept). 

Though I also find that having both `expires: Number` _and_ `expires: Date` is a duality we should be removing in a future version, simply because `Date` is powerful enough to easily cover the "days" scenario. In the past this was also causing confusion (e.g. why do we only support days, but not say hours or minutes?). Supporting `Date` and only `Date` would remove this ambiguity. The number == days case seemed sufficient and the simplest concept back in the jQuery days when the plugin had been initially written.. even support for `Date` was only added later on, when it was realized that not everyone would want to write cookies with expiration periods that could be represented in whole days. Allowing for fraction of days later was wrong imo (I find it rather unintuitive to specify `0.5` if I would want 12 hours until expiration), and was also further blurring the line between `Date` and `Number`...

Regarding comprehensibility of the test, I kind of agree, but whatever mechanism we would be building, in the end we'd have to test that the string being passed to `document.cookie` would contain the expiration in the expected format, I don't see any way around this (think integration test); that is, without rewriting more parts of the library with testability in mind (not against it, we could do this in version 3, also tackling the modularization). The way the code is structured still leans to the old jquery days and maybe it's time to overcome this.

All of this imo is beyond the scope of this supposedly tiny PR, where the aim was to fix a broken test.

Reply:
As mentioned, I would deprecate `expires: Number`, but keep `expires: Date`, thus not invent a new concept for how to deal with a date.

Reply:
PS: The test seems no more or less comprehensible than it was before, it's just fixed :)

Reply:
Cool LGTM, maybe something to rethink later.